### PR TITLE
Replace Guava with Java collections

### LINKED ISSUE #2212: Move to Java collections
Issue created to track desired changes:

- use Java collections instead of Guava
- Review streams use, and try to make them a bit more functional
- Make a better use of optionals

### PR DESCRIPTION
**Pull request by DavidRG13**
Closes #2212

### CODE REVIEW

**Comment by DavidRG13** [comment]
Please check if I'm right:
I changed the test as we can't ensure the order of the valid provider. We expect to be resolved as anything except the invalid, right?

File: core/cas-server-core-services/src/test/java/org/apereo/cas/authentication/DefaultMultifactorTriggerSelectionStrategyTests.java [hunk 1]
```diff
@@ -152,7 +153,7 @@ private static HttpServletRequest mockRequest(final String provider) {
 
     private static RegexRegisteredService mockService(final String... providers) {
         final DefaultRegisteredServiceMultifactorPolicy policy = new DefaultRegisteredServiceMultifactorPolicy();
-        policy.setMultifactorAuthenticationProviders(ImmutableSet.copyOf(providers));
+        policy.setMultifactorAuthenticationProviders(new HashSet<>(Arrays.asList(providers)));
```
Comment (mmoayyed) [inline core/cas-server-core-services/src/test/java/org/apereo/cas/authentication/DefaultMultifactorTriggerSelectionStrategyTests.java]: This is probably more like `Stream.of(providers).collect(Collectors.toSet())`? Or should be?

Comment (DavidRG13) [inline core/cas-server-core-services/src/test/java/org/apereo/cas/authentication/DefaultMultifactorTriggerSelectionStrategyTests.java]: what ever you prefer, choose one and make it the standard

File: core/cas-server-core-services/src/test/java/org/apereo/cas/authentication/DefaultMultifactorTriggerSelectionStrategyTests.java [hunk 2]
```diff
@@ -84,7 +85,7 @@ public void verifyRegisteredServiceTrigger() throws Exception {
         assertThat(strategy.resolve(VALID_PROVIDERS, null, mockService(MFA_PROVIDER_ID_2), null).orElse(null), is(MFA_PROVIDER_ID_2));
 
         assertThat(strategy.resolve(VALID_PROVIDERS, null, mockService(MFA_INVALID, MFA_PROVIDER_ID_1, MFA_PROVIDER_ID_2), null).get(),
-                is(MFA_PROVIDER_ID_1));
+                anyOf(is(MFA_PROVIDER_ID_2), is(MFA_PROVIDER_1)));
```
Comment (mmoayyed) [inline core/cas-server-core-services/src/test/java/org/apereo/cas/authentication/DefaultMultifactorTriggerSelectionStrategyTests.java]: This doesn't look right to me. The previous test specifically wants a match on MFA_PROVIDER_ID_1. So you probably should try to preserve order for the providers. Maybe use a LinkedHashSet?

File: core/cas-server-core-services/src/test/java/org/apereo/cas/authentication/DefaultMultifactorTriggerSelectionStrategyTests.java (hunk 1 above)
Comment (mmoayyed) [inline core/cas-server-core-services/src/test/java/org/apereo/cas/authentication/DefaultMultifactorTriggerSelectionStrategyTests.java]: Yup. Looked like we have a few of these `new HashSet<>(Arrays.asList...` patterns. I'll track this under a separate issue.

Comment (DavidRG13) [inline core/cas-server-core-services/src/test/java/org/apereo/cas/authentication/DefaultMultifactorTriggerSelectionStrategyTests.java]: Cool, feel free to assign it to me

**Comment by DavidRG13** [comment]
When this PR is fine to merge #2212 should be closed as we are not longer using Guava collections.

**Comment by mmoayyed** [comment]
Excellent. One test failure and we should be good on this one.

**Pull request merged**